lib/cluster-providers/generic-cluster-provider.ts

```diff
@@ -42,6 +43,66 @@ export interface GenericClusterProviderProps extends eks.ClusterOptions {
     }
 }
 
+export const managedNodeGroupContraints: ConstraintsType<ManagedNodeGroup> = {
+    /**
+     * id can be no less than 1 character long, and no greater than 63 characters long.
+     */
+    id: new StringConstraint(1, 63),
+    /**
+     * minSize has a maximum of 100 nodes per node group and can be as little as 0.
+     */
+    minSize: new NumberConstraint(0, 100),
```

Comment:
TODO: set the correct constraint and reference to the source, e.g. AWS account limits.